receiver/azuremonitorreceiver/config.go

```diff
@@ -235,22 +239,44 @@ type Config struct {
 	AppendTagsAsAttributes                  bool                          `mapstructure:"append_tags_as_attributes"`
 }
 
+const (
+	ServicePrincipal = "service_principal"
+	WorkloadIdentity = "workload_identity"
```

Comment:
These don't need to be public

```suggestion
	servicePrincipal = "service_principal"
	workloadIdentity = "workload_identity"
```

Reply:
thanks for the hint, updated the consts